sqlmesh/core/snapshot/evaluator.py

```diff
@@ -861,18 +905,20 @@ def _audit(
         blocking = audit_args.pop("blocking", None)
         blocking = blocking == exp.true() if blocking else audit.blocking
 
+        adapter = self.adapter(snapshot.model.gateway if snapshot.is_model else None)
```

Comment:
This is also repeated a lot, may be worth making a new method on snapshots that does this and do `snapshot.gateway` instead. Not sure which class in the snapshot hierarchy makes the most sense to have this.